src/alire/alire-origins-deployers-source_archive.adb

```diff
@@ -64,4 +48,18 @@ package body Alire.Origins.Deployers.Source_Archive is
       return Outcome_Success;
    end Deploy;
 
+   -------------------
+   -- Verify_Hashes --
+   -------------------
+
+   overriding
+   function Compute_Hash (This   : Deployer;
+                          Folder : String;
+                          Kind   : Hashes.Kinds) return Hashes.Any_Digest is
+      Archive_Name : constant String := This.Base.Archive_Name;
+      Archive_File : constant String := Dirs.Compose (Folder, Archive_Name);
+   begin
+      return Hashes.Digest (Hashes.Compute.Hash_File (Kind, Archive_File));
```

Comment:
Just to be sure, is this doing a hash of the archive before it is extracted?

Reply:
Yes